File: 2021/10/10/main.go
package main

func rangeBitwiseAnd(left int, right int) int {
	r := 0
	for left >= 1 && right >= 1 {
		if left == right {
			return left << r
		}
		left >>= 1
		right >>= 1
		r += 1
	}

	return 0
}

File: 2021/10/10/main.py
class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        r = 0
        while left >= 1 and right >=1:
            if left == right:
                return left << r
            left >>= 1
            right >>= 1
            r += 1

        return 0